src/core/etl/src/Flow/ETL/Pipeline/Optimizer/BatchSizeOptimization.php

```diff
@@ -27,26 +27,11 @@ final class BatchSizeOptimization implements Optimization
         PartitioningPipeline::class,
     ];
 
-    /**
-     * We can't use DbalLoader::class here because that would create a circular dependency between ETL and Adapters.
-     * All adapters requires ETL, but ELT does not require a single adapter to be present.
-     *
-     * @var array<class-string<Loader>>
-     */
-    private array $supportedLoaders = [
-        'Flow\ETL\Adapter\Doctrine\DbalLoader',
-        'Flow\ETL\Adapter\Elasticsearch\ElasticsearchPHP\ElasticsearchLoader',
-        'Flow\ETL\Adapter\Meilisearch\MeilisearchPHP\MeilisearchLoader',
-    ];
-
     /**
      * @param int<1, max> $batchSize
      */
-    public function __construct(private readonly int $batchSize = 1000, ?array $supportedLoaders = null)
+    public function __construct(private readonly int $batchSize = 1000)
```

Comment:
$batchSize is redundant, should come from the BatchingLoader method

Reply:
Yes, and no, we have a loader, but we have `Transformer` being allowed, so we should also have `BatchTransformer`, same asthe  loader?

Reply:
Added new transformer as well.

Reply:
whats the use case for optimizing batch size for transformers? I'm not sure if I fully understand which transformer would benefit from transforming in batch?

Reply:
yeah, I dont see how transformers could benefit from it, at least not now.